rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/resources/tests/NodeDetailSimple.spec.ts

```diff
@@ -0,0 +1,113 @@
+import { mount } from "@vue/test-utils";
+import NodeDetailsSimple from "../NodeDetailsSimple.vue";
+
+jest.mock("@/library/rundeckService", () => ({
+  getRundeckContext: jest.fn().mockReturnValue({
+    rdBase: "mockRdBase",
+    projectName: "test-project",
+  }),
+  url: jest.fn().mockReturnValue({ href: "mockHref" }),
+}));
+const mockAttributes = {
+  osFamily: "Linux",
+  osName: "Ubuntu",
+  osVersion: "20.04",
+  osArch: "x86_64",
+  description: "Test Node 1",
+  "ui:status:text": "Healthy",
+  "ui:status:icon": "fa-check",
+  "ns1:attr1": "value1",
+  "ns1:attr2": "value2",
+  "ns1:attr3": "value3",
+  "ns1:attr4": "value4",
+  "namespace:key1": "value1",
+  "namespace:key2": "value2",
+  key3: "value3",
+  key4: "value4",
+  key5: "value5",
+  key6: "value6",
+  username: "user",
+  hostname: "host",
+};
+const mountNodeDetailsSimple = async (propsData = {}) => {
+  const wrapper = mount(NodeDetailsSimple, {
+    props: {
+      useNamespace: true,
+      nodeColumns: true,
+      attributes: mockAttributes,
+      tags: ["Tag1", "Tag2"],
+      ...propsData,
+    },
+    global: {
+      mocks: {
+        $t: (msg) => msg,
+      },
+    },
+  });
+  await wrapper.vm.$nextTick();
+  return wrapper;
+};
+describe("NodeDetailsSimple Component", () => {
+  it("renders node attributes", async () => {
+    const wrapper = await mountNodeDetailsSimple();
+    const attributes = wrapper.findAll("td.value, td.key, .setting");
+    const attributeTexts = attributes.map((attr) => attr.text());
+    const expectedAttributes = {
+      osFamily: "Linux",
+      osName: "Ubuntu",
+      osVersion: "20.04",
+      osArch: "x86_64",
+      description: "Test Node 1",
+      "ui:status:text": "Healthy",
+      "ui:status:icon": "fa-check",
+      "ns1:attr1": "value1",
+      "ns1:attr2": "value2",
+      "ns1:attr3": "value3",
+      "ns1:attr4": "value4",
+    };
+    Object.entries(expectedAttributes).forEach(([key, value]) => {
+      const attributeFound = attributeTexts.some((text) =>
+        text.includes(value),
+      );
+      expect(attributeFound).toBe(true);
+    });
+  });
+  it("renders tags", async () => {
+    const wrapper = await mountNodeDetailsSimple();
+    const tags = wrapper.findAll(".label-muted");
+    expect(tags.length).toBe(2);
+    expect(tags.at(0).text()).toContain("Tag1");
+    expect(tags.at(1).text()).toContain("Tag2");
+  });
+  it("renders expandable attributes using class", async () => {
+    const wrapper = await mountNodeDetailsSimple();
+    const toggleButton = wrapper.find(".textbtn");
+    await toggleButton.trigger("click");
+    await wrapper.vm.$nextTick();
+    // Adjust the selector if necessary based on the actual HTML structure
+    const expandedAttributes = wrapper.findAll(".hover-action-holder");
+    expect(expandedAttributes.length).toBe(12);
+  });
+
+  it("renders correctly with useNamespace and nodeColumns set to true", async () => {
```

Comment:
The name of the test doesn't match what is being asserted. The node filter links that received the data-testids `node-attribute-link-username` and `node-attribute-link-hostname` will be rendered when the prop `useDefaultColumns` is true, independently of the values of `useNamespace` and `nodeColumns`.